src/lib/tpm.c

```diff
@@ -2478,6 +2478,51 @@ static CK_RV tpm_object_data_populate_ecc(TPM2B_PUBLIC *out_pub, tpm_object_data
     return rv;
 }
 
+static CK_RV check_tpm_ecc(ESYS_CONTEXT *context, tpm_key_data tpmdat) {
+
+    /* get ecc-curve capability from TPM*/
+    TPM2_CAP capability = TPM2_CAP_ECC_CURVES;
+    UINT32 property = TPM2_ECC_NIST_P192;
+    UINT32 counter = TPM2_MAX_ECC_CURVES;
+    TPMI_YES_NO more_data;
+    TPMS_CAPABILITY_DATA *capabilityData;
+
+    TSS2_RC rval = Esys_GetCapability (context,
+            ESYS_TR_NONE, ESYS_TR_NONE, ESYS_TR_NONE,
+            capability, property, counter,
+            &more_data, &capabilityData);
+
+    if (rval != TSS2_RC_SUCCESS) {
+        LOGE("Esys_GetCapability: 0x%x:", rval);
+        return CKR_GENERAL_ERROR;
+    }
+
+    if (!capabilityData || capabilityData->capability != capability) {
```

Comment:
FYI: if rval == TSS2_SUCCESS then capabilityData is guaranteed NOT to be NULL. This is the static analysis tool getting it wrong. I would say just drop an assert(capabilityData) after the check of rval..